lib/sbom/cyclonedx.go

```diff
@@ -53,3 +53,36 @@ func encodeCycloneDX(bom *cdx.BOM, f cdx.BOMFileFormat) encoderFn {
 		return cdx.NewBOMEncoder(w, f).Encode(bom)
 	}
 }
+
+func addCDXTool(bom *cdx.BOM, name, version string) {
+	if bom.Metadata == nil {
+		bom.Metadata = &cdx.Metadata{}
+	}
+
+	if bom.Metadata.Tools == nil {
+		bom.Metadata.Tools = &cdx.ToolsChoice{}
+	}
+
+	// Handle possibly existing, legacy "Tools" entry
+	if bom.Metadata.Tools.Tools != nil {
+		//nolint:staticcheck // Intentionally using a deprecated type in this case.
+		*bom.Metadata.Tools.Tools = append(*bom.Metadata.Tools.Tools, cdx.Tool{
```

Comment:
Should we check if `name` (and `version` for the components) already exists? I assume we don't want to clutter the tools array with `parlay` multiple times if the users runs different enrichers sequentially